.circleci/config.yml

```diff
@@ -0,0 +1,97 @@
+version: 2
+
+# --------------------------------
+# common : base
+# --------------------------------
+
+base_node: &base_node
+  docker:
+  - image: circleci/node:8.10.0-browsers
+
+# --------------------------------
+# common : steps
+# --------------------------------
+
+step_cache_restore_npm: &step_cache_restore_npm
+  name: 'Restore Cache: NPM Dependencies'
+  keys:
+  - node-v1-{{ .Branch }}-{{ checksum "package-lock.json" }}
+  - node-v1-{{ .Branch }}-
+  - node-v1-
+
+step_cache_save_npm: &step_cache_save_npm
+  name: 'Save Cache: NPM Dependencies'
+  paths:
+  - ~/usr/local/lib/node_modules  # location depends on npm version
+  key: node-v1-{{ .Branch }}-{{ checksum "package-lock.json" }}
+
+step_run_node_env_info: &step_run_node_env_info
+  name: 'Print Environment Info'
+  command: |
+    . ./.circleci/utils.sh
+    print_heading "environment versions:"
+    print_os_versions
+    print_node_versions
+    print_docker_versions
+
+step_run_update_npm: &step_run_update_npm
+  name: 'Update NPM To The most recent release'
+  shell: /bin/bash
+  command: |
+    sudo npm install -g npm@next
+
+step_run_install_npm_dependencies: &step_run_install_npm_dependencies
+  name: 'Install NPM Dependencies'
+  shell: /bin/bash
+  command: |
+    npm install
+
+step_lint_js: &step_lint_js
+  name: 'Lint JS'
+  command: |
+    npm run lint:js
+
+step_audit_dependencies: &step_audit_dependencies
+  name: 'Audit NPM Dependencies'
+  command: |
+    npm run audit
+
+step_test_coverage: &step_test_coverage
+  name: 'Tests Coverage'
+  command: |
+    npm run test:coverage
+
+step_test: &step_test
+  name: 'Tests'
+  command: |
+    npm run test
+
+
+# --------------------------------
+# jobs
+# --------------------------------
+
+jobs:
+  audit-linting-tests:
```

Comment:
so ATM in our other projects we name this according to what flow it applies to / what the goal is
eg.
  pull-request-job
  deploy-production-job
etc
instead of specifically what it does
what are your thoughts on this?
what would be the benefit of this approach?
I find this more descriptive:
```
  build:
    jobs:
      - pull-request-job:
          filters:
            branches:
              ignore:
                - master
                - develop
      - deploy-staging-job:
          context: Development
          filters:
            branches:
              only: develop
```
ie.
run general PR job on any branches except master or develop
and run the deploy to staging job on develop

is there a way to compose jobs that would make it useful to name them according to their steps?